ledger/ledger-api-common/src/test/suite/scala/com/digitalasset/platform/server/api/validation/DeduplicationPeriodValidatorSpec.scala

```diff
@@ -0,0 +1,49 @@
+package com.daml.platform.server.api.validation
+
+import java.time
+import java.time.Duration
+
+import com.daml.error.definitions.LedgerApiErrors.RequestValidation.InvalidDeduplicationPeriodField.ValidMaxDeduplicationFieldKey
+import com.daml.error.{ContextualizedErrorLogger, NoLogging}
+import com.daml.ledger.api.DeduplicationPeriod.DeduplicationDuration
+import com.daml.ledger.api.validation.ValidatorTestUtils
+import io.grpc.Status.Code.{FAILED_PRECONDITION, INVALID_ARGUMENT}
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.prop.TableDrivenPropertyChecks
+import org.scalatest.wordspec.AnyWordSpec
+
+class DeduplicationPeriodValidatorSpec
+    extends AnyWordSpec
+    with Matchers
+    with ValidatorTestUtils
+    with TableDrivenPropertyChecks {
+
+  private implicit val contextualizedErrorLogger: ContextualizedErrorLogger = NoLogging
+  private val maxDeduplicationDuration = time.Duration.ofSeconds(5)
+  private val deduplicationValidatorFixture = new ValidatorFixture(selfServiceErrorCodesEnabled =>
+    new DeduplicationPeriodValidator(ErrorFactories(selfServiceErrorCodesEnabled))
+  )
+
+  "not allow deduplication duration exceeding maximum deduplication duration" in {
```

Comment:
Does it make sense to split these 2 tests across 2 places? If so, maybe worth leaving a comment here.